src/scripts/underground/SeededRand.ts

```diff
@@ -10,7 +10,11 @@ class SeededRand {
     }
 
     public static seedWithDate(d: Date) {
-        this.state = Number((d.getFullYear() - 1900) * d.getDate() + 1000 * d.getMonth() + 100000 * d.getDate());
+        this.state = this.getDateSeed(d);
+    }
+
+    public static getDateSeed(d: Date): number {
+        return  Number((d.getFullYear() - 1900) * d.getDate() + 1000 * d.getMonth() + 100000 * d.getDate());
```

Comment:
Could we maybe change this to be `getSeedFromDate`
```suggestion
        this.state = this.getDateSeed(d);
    }

    public static getSeedFromDate(d: Date): number {
        return Number((d.getFullYear() - 1900) * d.getDate() + 1000 * d.getMonth() + 100000 * d.getDate());
```

Or just add a new method something like `seedWithDateHour`:
```suggestion
        this.state = Number((d.getFullYear() - 1900) * d.getDate() + 1000 * d.getMonth() + 100000 * d.getDate());
    }

    // hours specifies how many hours this seed should remain the same
    public static seedWithDateHour(d: Date, hours = 1): number {
        this.seedWithDate(d);
        const newHours = new Date(d - d % (GameConstants.HOUR * hours)).getHours();
        this.state += 1000000 * newHours;
```